tests/test_functional.py

```diff
@@ -390,17 +390,35 @@ def test_save_update_note(self, user, testapp, rodne_cislo):
         assert Note.query.count() == existing_notes + 1
 
     @pytest.mark.parametrize(
-        "rodne_cislo",
+        ("kwargs", "url_part"),
         [
-            "5005165649",
-            "5055172826",
-            "ahoj",
-            444,
+            ({"rc": 5005165649}, "detail"),
+            ({"rc": 5055172826}, "detail"),
+            ({"rc": "ahoj"}, "detail"),
+            ({"rc": 444}, "detail"),
+            ({"id": 0}, "batch_detail"),
+            ({"id": 999_999_999}, "batch_detail"),
+            ({"id": "ahoj"}, "batch_detail"),
+            ({"medal_slug": "aa"}, "award_prep"),
+            ({"medal_slug": "hello"}, "award_prep"),
+            ({"medal_slug": 0}, "award_prep"),
+        ],
+        ids=[
```

Comment:
May I ask what is the `ids` list for?

Reply:
ids is for name of test. If we dont use it, then output of pytest is not so detailed. Insted of concrete value we got `kwargs0` ,  `kwargs1` , etc...

Reply:
That sounds reasonable but do you think it's worth it to maintain two separated lists for it? I mean, if I see that a test for `kwargs1` fails, I can find the definition in the tests because the order should be preserved. Moreover, the default name contains `kwargsX` but also the name of the page like `detail` or `award_prep` which makes it even easier to find.

But, If you want to keep the ids as they are, I'm okay with that and I can imagine an easier way for the future maintenance: We can define the test data above the decorator and the let Python generate the ids for us in the decorator:
```python
    non_exist_arguments = (
        ({"rc": 5005165649}, "detail"),
        ({"rc": 5055172826}, "detail"),
        ({"rc": "ahoj"}, "detail"),
        ({"rc": 444}, "detail"),
        ({"id": 0}, "batch_detail"),
        ({"id": 999_999_999}, "batch_detail"),
        ({"id": "ahoj"}, "batch_detail"),
        ({"medal_slug": "aa"}, "award_prep"),
        ({"medal_slug": "hello"}, "award_prep"),
        ({"medal_slug": 0}, "award_prep"),
    )

    @pytest.mark.parametrize(
        ("kwargs", "url_part"),
        non_exist_arguments,
        ids=[str(arg) for arg in non_exist_arguments]
    )
```
I still think it's not necessary but this way you don't have to edit the data in two separated lists.

Reply:
I usually use a dictionary for this:
```python
testcases = {
   'one': {'input': 1, 'expected': '1'},
   'two': {'input': 2, 'expected': '2'},
}
@pytest.mark.parametrize('case_name', testcases)
def test_str(case_name):
    case = testcases[case_name]
    assert str(case['input']) == case['expected']
```

Reply:
My first attempt was with dict. I was not able to run it properly. I will try to look at dict sol deeper. If I will not be succesfull ', I will rewrite to case which were mentioned by Lumír. List comprehension of ids looks great for simplifying the code. Thanks for review...

Reply:
I mix suggestion of both. What do you think ?

I've removed test for batch_detail. Handling urls for `batch_detail` was changed in meantime. I have to study `@blueprint.get`....